packages/adblocker/src/filters/network.ts

```diff
@@ -1261,16 +1332,42 @@ export default class NetworkFilter implements IFilter {
     return this.getMask() & FROM_ANY;
   }
 
+  private getModifierOptionValue() {
+    return this.modifierOptionValue || '';
+  }
+
   public isRedirect(): boolean {
-    return this.redirect !== undefined;
+    return getBit(this.getMask(), NETWORK_FILTER_MASK.isRedirect);
   }
 
   public isRedirectRule(): boolean {
     return getBit(this.mask, NETWORK_FILTER_MASK.isRedirectRule);
   }
 
   public getRedirect(): string {
-    return this.redirect || '';
+    return this.getModifierOptionValue();
+  }
+
+  public isReplace(): boolean {
+    return getBit(this.getMask(), NETWORK_FILTER_MASK.isReplace);
+  }
+
+  public getHtmlModifier(): [RegExp, string] | null {
+    if (this.isReplace()) {
+      const [, rawRegexp, replacement, modifiers] = splitUnescaped(
+        this.getModifierOptionValue(),
+        '/',
+      );
+      const regexp = new RegExp(rawRegexp, modifiers);
+
+      return [regexp, replacement];
+    }
+
+    return null;
+  }
+
+  public isHtmlFilteringRule(): boolean {
```

Comment:
Thought: I wonder if it could make sense to also use the NetworkFilter abstraction to represent the ^script-text html filters?

Reply:
I was also thinking a similar system to that. However, I don't know if it's possible to make a proper regexp filtering system with streaming data.

Reply:
The current implementation already supports regexps from what I can tell.

For example this filter (already supported):
```
##^script:has-text(/innerHTML.*appendChild/)
```

Would be equivalent to (if my understanding is correct):
```
$replace=/innerHTML.*appendChild//
```

So the current mechanism needs to be extended but it seems doable without changing to a different framework.

Reply:
The current implementation does buffer the all of the script tag until they're finished as I know. This means we know the start and end of the part. However, `$replace` is performed in full content which means we might need to know full text.